src/ready/apis/train_mobious.py

```diff
@@ -326,17 +372,74 @@ def main(args):
+                 for j, data in enumerate(validation_loader, 1):
+                    
+                    images, labels = data
+                    if cuda_available:
+                        images = images.cuda()
+                        labels = labels.cuda()
+
+                    optimizer.zero_grad()
+                    output = model(images)
+                    print(f"output.size() {output.size()};\
+                    type(output): {type(output)};\
+                    pred.type: {output.type()} ")
+                    # torch.Size([batch_size_, 4, 400, 640]);
+                    # <class 'torch.Tensor'>;
+                    # torch.cuda.FloatTensor
+
+                    loss = loss_fn(output, labels)
+                    
+                    batch_metrics = evaluate(output, labels)
+
+                    for key, value in batch_metrics.items():
+                        # print(f"{key}: {value:.4f}")
+                        validation_performance[key] += value * len(images) # weighted by batch size
+
+                    num_samples += len(images)
+                    validation_running_loss += loss.item()
+
+                    # Log every X batches
+                    if j % 50 == 0 or j == 1:
+                        print(f"Validation: Loss at {j} mini-batch {loss.item():.4f}")
+                    # TODO
+                    #                sanity_check(trainloader, model, cuda_available)
+                    #                save_checkpoint(
+                    #                    {
+                    #                        "epoch": run_epoch,
+                    #                        "state_dict": model.state_dict(),
+                    #                        "optimizer": optimizer.state_dict(),
+                    #                    },
+                    #                    "models/o.pth",
+                    #                )
+                    #
+                    # if j == 300:
+                    #     break
+                    # # performance[key].append(average_metric)
+
+        training_epoch_loss = calculate_epoch_loss(training_running_loss, num_samples)
+        validation_epoch_loss = calculate_epoch_loss(validation_running_loss, num_samples)
+        
+        training_loss_values.append(training_epoch_loss)
+        validation_loss_values.append(validation_epoch_loss)
+        print(f"\nTraining epoch loss: {training_epoch_loss:.4f}")
+        print(f"Validation epoch loss: {validation_epoch_loss:.4f}\n")
+        
+        print(f"Training Metrics:")
+        for key in performance_metrics_labels: 
+            training_performance[key] /= num_samples
+            print(f"Average {key} @ epoch: {training_performance[key]:.4f}")
+        
+        print(f"\nValidation Metrics:")
+        for key in performance_metrics_labels:     
+            validation_performance[key] /= num_samples
+            print(f"Average {key} @ epoch: {validation_performance[key]: .4f}")
```

Comment:
can we mention how to plot training and validation metrics as in plot losses at https://github.com/oocular/ready/tree/93-splitting-data-into-sets/data/mobious/models. Maybe also you can add descriptions within the [README](https://github.com/oocular/ready/blob/93-splitting-data-into-sets/data/mobious/models/README.md)